bot/exts/moderation/infraction/_scheduler.py

```diff
@@ -434,9 +443,19 @@ async def deactivate_infraction(
         try:
             # Mark infraction as inactive in the database.
             log.trace(f"Marking infraction #{id_} as inactive in the database.")
+
+            data = {"active": False, "reason": ""}
+
+            if pardon_reason is not None:
+                # Append pardon reason to infraction in database.
+                if (punish_reason := infraction["reason"]) is not None:
+                    data["reason"] = punish_reason + " | "
+
+                data["reason"] += f"Pardoned: {pardon_reason}"
+
             await self.bot.api_client.patch(
                 f"bot/infractions/{id_}",
-                json={"active": False}
+                json=data
             )
```

Comment:
Hmm, now if `pardoned_reason` is none it'll try to update `reason` to be an empty string.

Reply:
Fixed - removed the "reason" field from data so it'll just have `{"active": False}` if `pardon_reason` is `None`

Reply:
Now it'll error if `data["reason"] += f"Pardoned: {pardon_reason}"` is hit without reason being set first (if the reason is currently `None`) 😅 

I think setting `data["reason"] = ""` first in `if pardon_reason is not None:` would fix it.

Reply:
Added.